R/regex_subset_linter.R

```diff
@@ -72,7 +72,7 @@ regex_subset_linter <- function() {
       grep_expr,
       source_expression = source_expression,
       lint_message =
-        "Prefer grep(pattern, x, ..., value = TRUE) over x[grep(pattern, x, ...)] and x[grepl(pattern, x, ...)].",
+        "Prefer grepv(pattern, x, ...) over x[grep(pattern, x, ...)] and x[grepl(pattern, x, ...)].",
```

Comment:
Can we maybe customize this error message based on the R version since lintr is still compatible with R 4.0 to 4.4, which don't know about `grepv()`?

Reply:
I wouldn't go that way, since the process running {lintr} can be different from the DESCRIPTION dependency, or any other number of incompatibilities between the report and the actual intention.

WDYT about just adding

> Use {backports} if your code might also run on R versions before 4.5.0.

Reply:
Sounds good in theory but we would have to wait until https://github.com/r-lib/backports/pull/88 is merged and a new version of backports released.

Maybe the safest option is to be more verbose:

> Prefer grepv(pattern, x, ...), or grep(pattern, x, value = TRUE) if your code might also run on R versions before 4.5.0, over x[grep(pattern, x, ...)] and x[grepl(pattern, x, ...)].

Reply:
Adjusted, PTAL.